misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/ProgramAnalyzerNodeVisitor.java

```diff
@@ -375,1133 +196,165 @@ public void visit(FunctionDefinitionNode functionDefinitionNode) {
-
-    }
-
-    @Override
-    public void visit(TransactionStatementNode transactionStatementNode) {
-
-    }
-
-    @Override
-    public void visit(RollbackStatementNode rollbackStatementNode) {
-
-    }
-
-    @Override
-    public void visit(RetryStatementNode retryStatementNode) {
-
-    }
-
-    @Override
-    public void visit(CommitActionNode commitActionNode) {
-
-    }
-
-    @Override
-    public void visit(TransactionalExpressionNode transactionalExpressionNode) {
-
-    }
-
-    @Override
-    public void visit(ByteArrayLiteralNode byteArrayLiteralNode) {
-
-    }
-
-    @Override
-    public void visit(XMLFilterExpressionNode xMLFilterExpressionNode) {
-
-    }
-
-    @Override
-    public void visit(XMLStepExpressionNode xMLStepExpressionNode) {
-
-    }
-
-    @Override
-    public void visit(XMLNamePatternChainingNode xMLNamePatternChainingNode) {
-
-    }
-
-    @Override
-    public void visit(XMLAtomicNamePatternNode xMLAtomicNamePatternNode) {
-
-    }
-
-    @Override
-    public void visit(TypeReferenceTypeDescNode typeReferenceTypeDescNode) {
-
-    }
-
-    @Override
-    public void visit(MatchStatementNode matchStatementNode) {
-
-    }
-
-    @Override
-    public void visit(MatchClauseNode matchClauseNode) {
-
-    }
-
-    @Override
-    public void visit(MatchGuardNode matchGuardNode) {
-
-    }
-
-    @Override
-    public void visit(DistinctTypeDescriptorNode distinctTypeDescriptorNode) {
-
-    }
-
-    @Override
-    public void visit(ListMatchPatternNode listMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(RestMatchPatternNode restMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(MappingMatchPatternNode mappingMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(FieldMatchPatternNode fieldMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(ErrorMatchPatternNode errorMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(NamedArgMatchPatternNode namedArgMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(MarkdownDocumentationNode markdownDocumentationNode) {
+    public void visit(VariableDeclarationNode variableDeclarationNode) {
 
-    }
+        Optional<Symbol> symbol = model.symbol(variableDeclarationNode);
+        if (symbol.isEmpty()) {
+            return;
+        }
 
-    @Override
-    public void visit(MarkdownDocumentationLineNode markdownDocumentationLineNode) {
+        if (variableDeclarationNode.initializer().isEmpty()) {
+            return;
+        }
 
-    }
+        if (isClientObject(symbol.get())) {
+            resolveEndPoint(symbol.get(), variableDeclarationNode.initializer().get());
+        }
 
-    @Override
-    public void visit(MarkdownParameterDocumentationLineNode markdownParameterDocumentationLineNode) {
+        if (isClientObject(symbol.get()) || isRecordObject(symbol.get())) {
+            Optional<ExpressionNode> expressionNode = variableDeclarationNode.initializer();
```

Comment:
Get this to a variable. We have done the `isEmpty` check above and we do not need to repeat. Also  after this refactor, you can merge the two checks